simd/src/Kokkos_SIMD_AVX512.hpp

```diff
@@ -30,6 +30,18 @@
     "Kokkos_SIMD_AVX512.hpp must be included before Kokkos_SIMD_Common_Math.hpp!"
 #endif
 
+// FIXME: Some intrinsics we are using are not marked constexpr in GCC
+// prior to some version. 8.5 fails 9.5 works.
+#ifdef KOKKOS_COMPILER_GNU
+#if KOKKOS_COMPILER_GNU < 950
+#define KOKKOS_IMPL_SIMD_CONSTEXPR
+#endif
+#endif
+
+#ifndef KOKKOS_IMPL_SIMD_CONSTEXPR
+#define KOKKOS_IMPL_SIMD_CONSTEXPR constexpr
```

Comment:
Should we `#undef` this at the end of the file?

Reply:
Ping

Reply:
Depends on https://github.com/kokkos/kokkos/pull/7486.

Reply:
`constexpr` changes won't be part of #7486 (I will cherry-pick 7445d06bc7a7239f66ca59dc135d989287150d40 for reliable AVX512 testing).
I plan on revisiting constexpr-ness of simd interfaces in a separate PR later.
So, I think this PR could be closed for now.